### DISCUSSION TITLE
service unavaliable during the rolling update

### DISCUSSION THREAD

**Discussion by stardustman** [linkerd/linkerd2#8516] [Help]
hello, I use the Linkerd in production env, I just use `linkerd.io/inject: enabled` to mesh the gRPC servcie pod, but during the rolling update, the service is unavaliable, how can I resolve it? Is there a check list I can perform to solve it?

**Comment by cpretzer** [comment]
Hi @stardustman sorry to hear that there are some issues with the rolling update. 

Can you describe more about how the service is unavailable? In a rolling update, I'd expect an existing pod to continue to handle requests until the new pod is ready to handle the requests. Then the traffic will seamlessly be handed off to the new pod while the old pod is removed. 

Have you checked the logs for the service and the Linkerd proxy? Are there any interesting messages in the `kubectl get ev` output?

Have you checked whether the same behavior occurs when the service isn't injected with the Linkerd proxy?
